src/viking.js

```diff
@@ -43,29 +41,43 @@ class Saxon extends Soldier {
     }
     receiveDamage(theDamage){
         this.health -= theDamage;
-        if (this.health> 0){
-            return `A Saxon has received ${theDamage} points of damage`
-        }
-        else {
-            return `A Saxon has died in combat`
-        }
+        if (this.health> 0) return `A Saxon has received ${theDamage} points of damage`
+        
+        else return `A Saxon has died in combat`
+        
 } }
 
 class War {
-    constructor()
+    vikingArmy = []
+    saxonArmy = []
+
+    addViking(viking){
+    this.vikingArmy.push(viking)
+    }
+    addSaxon(saxon){
+    this.saxonArmy.push(saxon)
+    }
+    vikingAttack(){
```

Comment:
This iteration is not passing this teste `"should return result of calling receiveDamage() of a Saxon with the strength of a Viking"`.
This is happening because you are returning the `this.saxonArmy.splice(saxon, 1);` inside the `if`, and the result of the attack is being returned inside an `else`. 
This means that the code will run and if `(!(this.saxonArmy[saxon].health > 0))` is true you will remove the dead saxon from the array and it will stop there, so the `return valor` will never be read.
To solve this you should do something like this:

```
(...)

  vikingAttack() {
    let viking = Math.floor(Math.random() * this.vikingArmy.length);
    let saxon = Math.floor(Math.random() * this.saxonArmy.length);
    let valor = this.saxonArmy[saxon].receiveDamage(
      this.vikingArmy[viking].strength
    );

    if (!(this.saxonArmy[saxon].health > 0)) {
      this.saxonArmy.splice(saxon, 1);
    }

    return valor;
  }

(...)
```